beginend.el

```diff
@@ -232,6 +232,32 @@ BEGIN-BODY and END-BODY are two `progn' expressions passed to respectively
   (progn
     (prodigy-last)))
 
+;; Copy/pasted from expand-region
+(defun beginend--point-is-in-comment-p ()
+  "Return non-nil if point is in comment."
+  (or (nth 4 (syntax-ppss))
+      (memq (get-text-property (point) 'face)
+            '(font-lock-comment-face font-lock-comment-delimiter-face))))
```

Comment:
Actually, I've just tried a simple regexp solution `(looking-at "\\s-*\\(\\s<.*?\\)?$")` but this fails exactly in cases I mentioned above... for example C `/*` style comments because `/` there is punctuation not a comment starter (syntax class `<` matched by `\\s<`).  So I guess the check above could be adapted (but also one would need to make sure all whitespace is skipped from `bol` to land to the first character, which is either in comment already or has a comment-starting syntax properties).

As for `-when-let` we can simply macroexpand it (use https://github.com/joddie/macrostep if you don't already, it's awesome).

I also use `sp--syntax-ppss` which is a version with better caching behaviour, can be replaced with the built-in, the performance here will not be critical.

Should I go ahead and make the change?